terraso_backend/apps/graphql/schema/schema.graphql

```diff
@@ -1527,6 +1528,96 @@ String, Boolean, Int, Float, List or Object.
 """
 scalar GenericScalar
 
+"""Soil ID algorithm queries."""
+type SoilID {
+  locationBasedSoilMatches(latitude: Float!, longitude: Float!): LocationBasedSoilMatches!
+  dataBasedSoilMatches(latitude: Float!, longitude: Float!, data: SoilIDInputData!): DataBasedSoilMatches!
```

Comment:
I'm a little concerned about the complexity of the SoilIDInputData type -- would it make sense to take in a site ID instead, and then look up the necessary data in the query resolver?

I definitely think it's good to have the underlying implementation be decoupled from any notion of sites, but I'm pretty sure the only use-case we have for data-based lookups will be sourcing this from sites, right? And so it might be better for the UI to not have to transform and send this data over the wire when it's already available to the backend.

(If I'm wrong about any of this, please correct me!)

Reply:
your understanding is all correct, and it's a good question. this is my thinking for why to construct the API this way:
- in either case, we need to implement the transformation from site to soil id input data in one place (either in the frontend or the backend), so it's not necessarily more work either way
- the amount of data sent over the wire is very small in practice (it's a list of length ~10 of json objects with ~5 fields)
- i agree that for the capri feature set, it seems a bit nicer to just take a site ID. but there are multiple important future design goals in the app which would require us to eventually do this work client-side anyway:
  - running the algorithm offline
  - running the algorithm for non-signed in users who don't want to persist their data on our backend

i try to stay away from pre-designing things, but in this case where the trade-off seems small and the future considerations are pretty concrete and important it seemed like the better choice. thoughts?

Reply:
Your reasoning here makes sense and I think that ultimately I'd like to have both versions of the endpoint for exactly the reasons given here. I think I'm coming at this from a slight preference for backend-side transformation of data for the reason that our backend seems more well-tested / easier to nail down data conditions, but it's ultimately a decision of low consequence (we can always add a backend API built on top of this existing one if it proves too cumbersome to do the transformation on the frontend.)